esi_ui/api/esi_api.py

```diff
@@ -2,81 +2,122 @@
 
 from django.conf import settings
 
-from ironicclient.v1.client import Client as _Ironicclient
 from esi import connection
+from esi.lib import nodes
 
 from openstack import config
 from horizon.utils.memoized import memoized  # noqa
 
+
 DEFAULT_OPENSTACK_KEYSTONE_URL = 'http://127.0.0.1/identity/v3'
 
+
 @memoized
 def get_session_from_token(token):
     auth_url = getattr(settings, 'OPENSTACK_KEYSTONE_URL', DEFAULT_OPENSTACK_KEYSTONE_URL)
     region = config.get_cloud_region(load_yaml_config=False,
                                      load_envvars=False,
                                      auth_type='token',
                                      token=token,
-                                     auth_url=auth_url,
-                                     )
+                                     auth_url=auth_url)
     return region.get_session()
 
+
 @memoized
 def esiclient(token):
     return connection.ESIConnection(session=get_session_from_token(token))
 
-@memoized
-def ironicclient(token):
-    return _Ironicclient(session=get_session_from_token(token))
-
 
 def node_list(request):
-    token = request.user.token.id
+    connection = esiclient(token=request.user.token.id)
+
+    def node_list_full_info(connection):
+        """Get a list of nodes with information from Ironic and ESI-Leap
+
+        :param connection: An ESI connection
+        :type connection: :class:`~esi.connection.ESIConnection`
+        :returns: A list of dictionaries containing baremetal and lease information
+        """
+
+        esi_nodes = []
+        baremetal_nodes = []
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            f1 = executor.submit(connection.lease.nodes)
+            f2 = executor.submit(connection.baremetal.nodes, details=True)
+            esi_nodes = list(f1.result())
+            baremetal_nodes = list(f2.result())
+
+        esi_nodes.sort(key=lambda node: node.name)
+        baremetal_nodes.sort(key=lambda node: node.name)
```

Comment:
I still don't think this is a great way to correspond the various list of nodes. Nodes in Ironic don't technically need a unique name, and if the behavior of the services change then it's possible that the list of nodes will actually be different. It's much much better to simply match by node ID.